app/routes/weetsRoutes.js

```diff
@@ -1,13 +1,21 @@
 // eslint-disable-next-line new-cap
 const router = require('express').Router();
 const weetInteractor = require('../interactors/weetInteractor');
-const { getWeet } = require('../controllers/weetsController');
 const protectedRoute = require('../middlewares/protectedRoute');
 const asyncWrapper = require('../utils/asyncWrapper');
 const weetMapper = require('../mappers/weetMapper');
 const httpCodes = require('../constants/httpCodes');
+const validateSchema = require('../middlewares/validateSchema');
+const { paginationSchema } = require('../validations');
 
-router.get('/', [], getWeet);
+router.get(
+  '/',
+  [protectedRoute, validateSchema(paginationSchema)],
+  asyncWrapper(async (req, res) => {
+    const { results, pagination } = await weetInteractor.getWeets(req.query);
```

Comment:
You should not call an interactor from a router, neither have a middleware definition. Move this back to the controller :stuck_out_tongue: